include/engine/datafacade/internal_datafacade.hpp

```diff
@@ -727,20 +739,18 @@ class InternalDataFacade final : public BaseDataFacade
          * both forward and reverse segments along the same bi-
          * directional edge. The m_geometry_indices stores
          * refences to where to find the beginning of the bi-
-         * directional edge in the m_geometry_list vector.
+         * directional edge in the m_geometry_node_list vector.
          * */
-        const unsigned begin = m_geometry_indices.at(id);
-        const unsigned end = m_geometry_indices.at(id + 1);
+        const signed begin = m_geometry_indices.at(id);
+        const signed end = m_geometry_indices.at(id + 1);
```

Comment:
Where is this change comming from? In general `signed` is quite unusual in C++, I would actually recommend doing `auto` everywhere in these functions.